scipy/stats/stats.py

```diff
@@ -2276,7 +2277,12 @@ def obrientransform(*args):
             raise ValueError('Lack of convergence in obrientransform.')
 
         arrays.append(t)
+        sLast = a.shape
 
+    if sLast:
+        for arr in arrays[:-1]:
+            if sLast != arr.shape:
+                return np.array(arrays, dtype=object)
```

Comment:
this is the only real code change (in code, not in a  `test*.py` file), and is consistent with the docstring